### ISSUE TITLE
customized labelling of tree nodes

### ISSUE THREAD

**Issue by chrisclarkson100** [wannesm/dtaidistance#9]
Hi your package is very interesting to me and I would like to use it to visualise my own data.
I would like to plot a dendrogram with each of the node leaves labelled according to an external label. 

However I cannot seem to find a way to do this....

```
import numpy as np
import pandas as pd
dataframe = pd.read_csv('sample.csv',header=None)
dataframe.head(6)

                                                                           label for nodes
0  chr13_110718378-110719378.txt    _2.441430_  207.521542  163.575804   ......    2
1     chr2_96278196-96279196.txt   43.223219  242.530287  168.090298     ......    1
2   chr4_140084844-140085844.txt  237.444590  155.823012  249.811496  ......    3
3    chr10_71267774-71268774.txt  232.878508  139.246943  225.676080   .......   3
4    chr14_86309018-86310018.txt  131.655232  248.406099   67.069647   .......    2
5     chr3_97076527-97077527.txt  129.814476    0.000000  204.337600   ........     1

df=dataframe.values
labels=df[:,-1] #labels variable
df=df[:,:-1]
df= df[:,1:]

from dtaidistance import dtw
from dtaidistance import dtw_visualisation as dtwvis
from dtaidistance import clustering
model1 = clustering.Hierarchical(dtw.distance_matrix_fast, {})
model2 = clustering.HierarchicalTree(model1)
model3 = clustering.LinkageTree(dtw.distance_matrix_fast, {})
cluster_idx = model3.fit(np.matrix(df,dtype=np.double))
```
Hence my question is if there is a way to label the tree nodes and still have the nice plot formats that your package produces? 

![test](https://user-images.githubusercontent.com/8440573/42817935-7b63633c-89c7-11e8-9ac2-8af53d2a73cd.png)

**Comment by wannesm** [comment]
You can pass an axes object that you created yourself to the plot function. When creating that figure, you have access to all settings matplotlib offers. For example, a taller image can be achieve with:

```
fig, ax = plt.subplots(nrows=1, ncols=2, figsize=(10, 20))
model.plot(hierarchy_fn, axes=ax)
```

I have also pushed a new version that exposes the parameters for the clustering plotting function (version 1.1.1).  You can now, for example, have a custom label and position it before the series:

```
model.plot(hierarchy_fn, axes=ax, show_ts_label=lambda idx: "ts-" + str(idx),
           show_tr_label=True, ts_label_margin=-10,
           ts_left_margin=10, ts_sample_length=1)
```

**Comment by chrisclarkson100** [comment]
Hi thank you for getting back to me so quickly: I have installed the latest push and tried it.
I should have explained what I needed better- what I wanted was something similar to the following:

```
dataframe = pd.read_csv('sample.csv',header=None)
dataframe.head(6)

                                                                           label for nodes
0  chr13_110718378-110719378.txt    _2.441430_  207.521542  163.575804   ......    2
1     chr2_96278196-96279196.txt   43.223219  242.530287  168.090298     ......    1
2   chr4_140084844-140085844.txt  237.444590  155.823012  249.811496  ......    3
3    chr10_71267774-71268774.txt  232.878508  139.246943  225.676080   .......   3
4    chr14_86309018-86310018.txt  131.655232  248.406099   67.069647   .......    4
5     chr3_97076527-97077527.txt  129.814476    0.000000  204.337600   ........     1

```

see the innermost/ outermost column is what I intended to be a label eg.

```
labels=df[:,-1] #labels variable
df=df[:,:-1]
df= df[:,1:]
```

Hence I tried the following and got an error:

```
model3.plot(cluster_idx, axes=ax, show_ts_label=list(labels),
           show_tr_label=True, ts_label_margin=-10,
           ts_left_margin=10)
Traceback (most recent call last):
  File "<stdin>", line 5, in <module>
  File "/home/bsxcto/miniconda3/lib/python3.6/site-packages/dtaidistance/clustering.py", line 195, in plot
    raise AttributeError("Unknown type for show_ts_label: {}".format(type(show_ts_label)))
AttributeError: Unknown type for show_ts_label: <class 'list'>
```
**Hoping for something like:**
![screenshot from 2018-07-17 17-01-52](https://user-images.githubusercontent.com/8440573/42831365-0f9931f8-89e6-11e8-9532-c997f531db12.png)

**Comment by wannesm** [comment]
It should be a callable object, not a subscriptable one. Try:

mylabels = list(labels)
model3.plot(cluster_idx, axes=ax, show_ts_label=lambda idx: mylabels[idx],
           show_tr_label=True, ts_label_margin=-10,
           ts_left_margin=10)

I have added support for lists in the version on github.

wannes

**Comment by wannesm** [comment]
Released as v1.1.2
